cate/core/opimpl.py

```diff
@@ -407,45 +488,51 @@ def subset_spatial_impl(ds: xr.Dataset,
     not the polygon itself be masked with NaN.
     :return: Subset dataset
     """
+    monitor.start('Subset', 10)
+    # Validate input
+    try:
+        polygon = PolygonLike.convert(region)
+    except BaseException as e:
+        raise e
 
-    if not region.is_valid:
-        # Heal polygon, see #506 and Shapely User Manual
-        region = region.buffer(0)
+    extents, explicit_coords = get_extents(region)
 
-    # Get the bounding box
-    lon_min, lat_min, lon_max, lat_max = region.bounds
+    lon_min, lat_min, lon_max, lat_max = extents
 
     # Validate the bounding box
     if (not (-90 <= lat_min <= 90)) or \
             (not (-90 <= lat_max <= 90)) or \
             (not (-180 <= lon_min <= 180)) or \
             (not (-180 <= lon_max <= 180)):
-        raise ValueError('Provided polygon extent outside of geospatial'
+        raise ValueError('Provided polygon extends outside of geospatial'
```

Comment:
Please use "extent", see http://www.dictionary.com/browse/extent

Reply:
Ah... I don't agree. :)
The depends of you want a statement, or a description. But even if it's a statement, just 'extent' would be wrong-ish. So it's either:
'Given polygon extent is outside of bounds'

Or with a verb (http://www.dictionary.com/browse/extend)
'Provided polygon extends outside of bounds'